hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -180,6 +198,16 @@ private boolean initializeIteration() {
       return false;
     }
 
+    this.clusterCapacity = 0L;
+    this.clusterUsed = 0L;
+    this.clusterRemaining = 0L;
+
+    this.selectedContainers = new HashSet<>();
+    this.overUtilizedNodes = new ArrayList<>();
+    this.underUtilizedNodes = new ArrayList<>();
+    this.withinThresholdUtilizedNodes = new ArrayList<>();
+    this.unBalancedNodes = new ArrayList<>();
```

Comment:
Where do you suggest these lists should be initialized then?

Reply:
`clusterCapacity`, `clusterUsed` and `clusterRemaining ` will be renewed in `calculateAvgUtilization` at every iteration, seems no need to set the to 0 here specially.

and  other variables can be renewed using `clear()` , no need to create a new object every time

Reply:
Since one purpose of `intitalizeIteration` is to initialize the fields necessary for balancing, it seems clearer to initialize these values here instead of `calculateAvgUtilization`. Also since those lists need to be initialized at least once, we need to create them here. Instead, if this initialization is moved to the constructor, we will need to construct every time we want to restart balancer.